src/toil/batchSystems/parasol.py

```diff
@@ -238,9 +247,64 @@ def getRescueBatchJobFrequency(cls):
         making it expensive. 
         """
         return 5400 #Once every 90 minutes
+    def updatedJobWorker(self):
+        """We use the parasol results to update the status of jobs, adding them
+        to the list of updated jobs.
+
+        Results have the following structure.. (thanks Mark D!)
+
+        int status;    /* Job status - wait() return format. 0 is good. */
+        char *host;    /* Machine job ran on. */
+        char *jobId;    /* Job queuing system job ID */
+        char *exe;    /* Job executable file (no path) */
+        int usrTicks;    /* 'User' CPU time in ticks. */
+        int sysTicks;    /* 'System' CPU time in ticks. */
+        unsigned submitTime;    /* Job submission time in seconds since 1/1/1970 */
+        unsigned startTime;    /* Job start time in seconds since 1/1/1970 */
+        unsigned endTime;    /* Job end time in seconds since 1/1/1970 */
+        char *user;    /* User who ran job */
+        char *errFile;    /* Location of stderr file on host */
+
+        plus you finally have the command name..
+        """
+        resultsFiles = set()
+        resultsFileHandles = []
+        while self.running:
+            #Look for any new results files that have been created, and open them
+            newResultsFiles = set(os.listdir(self.parasolResultsDir)).difference(resultsFiles)
+            for newFile in newResultsFiles:
+                resultsFiles.add(newFile)
+                newFilePath = os.path.join(self.parasolResultsDir, newFile)
+                resultsFileHandles.append(open(newFilePath, 'r'))
+
+            for fileHandle in resultsFileHandles:
+                line = fileHandle.readline()
```

Comment:
Actually, this code is sensitive to the buffering pattern on the writing side:

```python
>>> w=open('foo','w')
>>> r=open('foo')
>>> r.readline()
''
>>> w.write('bar\n')
>>> r.readline()
''
>>> w.flush()
>>> r.readline()
'bar\n'
>>> w.write('bar2') # no newline at the end
>>> w.flush()
>>> w.write('\n') # now write newline
>>> r.readline() # returns partial line, not good
'bar2'
>>> w.write('bar3\n')
>>> w.flush()
>>> r.readline()
'\n'
>>> r.readline()
'bar3\n'
>>> w.close()
>>> r.close()
```

Not quite sure what the best way to handle this is. Probably just the assertion so we know early when this happens.